crates/pop-parachains/src/new_pallet.rs

```diff
@@ -2,55 +2,98 @@
+	/// The name of the pallet
 	pub name: String,
+	/// The authors of the pallet
 	pub authors: String,
+	/// The pallet description
 	pub description: String,
+	/// A `bool` indicating if the pallet is contained in a workspace
 	pub pallet_in_workspace: bool,
+	/// A `bool` indicating if the user wanna use the advanced mode
+	pub pallet_advanced_mode: bool,
+	/// A `bool` indicating if the template must include a default config for the pallet.
+	pub pallet_default_config: bool,
+	/// A `Vec` indicating which of the types defined in `TemplatePalletConfigCommonTypes` should
+	/// be included in the template.
+	pub pallet_common_types: Vec<TemplatePalletConfigCommonTypes>,
+	/// A `Vec` containing which type of storages are used and their names.
+	pub pallet_storage: Vec<TemplatePalletStorageTypes>,
+	/// A `bool` indicating if the template should include a genesis config
+	pub pallet_genesis: bool,
+	/// A `bool` indicating if the template should include a custom origin
+	pub pallet_custom_origin: bool,
 }
 /// Create a new pallet from a template.
 ///
 /// # Arguments
 ///
 /// * `path` - location where the pallet will be created.
 /// * `config` - customization values to include in the new pallet.
-pub fn create_pallet_template(
-	path: Option<String>,
-	config: TemplatePalletConfig,
-) -> Result<(), Error> {
-	let target = resolve_pallet_path(path)?;
+pub fn create_pallet_template(target: PathBuf, config: TemplatePalletConfig) -> Result<(), Error> {
 	let pallet_name = config.name.clone();
 	let pallet_path = target.join(pallet_name.clone());
+
 	sanitize(&pallet_path)?;
-	generate_pallet_structure(&target, &pallet_name)?;
+	generate_pallet_structure(&target, &pallet_name, &config)?;
 
 	render_pallet(pallet_name, config, &pallet_path)?;
 	Ok(())
 }
 
 /// Generate a pallet folder and file structure
-fn generate_pallet_structure(target: &PathBuf, pallet_name: &str) -> Result<(), Error> {
+fn generate_pallet_structure(
+	target: &PathBuf,
+	pallet_name: &str,
+	config: &TemplatePalletConfig,
+) -> Result<(), Error> {
 	use fs::{create_dir, File};
-	let (pallet, src) = (target.join(pallet_name), target.join(pallet_name.to_string() + "/src"));
+	let (pallet, src, pallet_logic, tests) = (
+		target.join(pallet_name),
+		target.join(pallet_name.to_string() + "/src"),
+		target.join(pallet_name.to_string() + "/src/pallet_logic"),
+		target.join(pallet_name.to_string() + "/src/tests"),
+	);
 	create_dir(&pallet)?;
 	create_dir(&src)?;
 	File::create(format!("{}/Cargo.toml", pallet.display()))?;
 	File::create(format!("{}/lib.rs", src.display()))?;
 	File::create(format!("{}/benchmarking.rs", src.display()))?;
 	File::create(format!("{}/tests.rs", src.display()))?;
 	File::create(format!("{}/mock.rs", src.display()))?;
-	File::create(format!("{}/weights.rs", src.display()))?;
+	if config.pallet_advanced_mode {
+		create_dir(&pallet_logic)?;
+		create_dir(&tests)?;
+		File::create(format!("{}/pallet_logic.rs", src.display()))?;
```

Comment:
`pallet_logic` and `try_state` files I am not sure if makes sense to add them.
@brunopgalvao  any thoughts?

Reply:
This is just to create a modular template, so the advanced developer doesn't have to separate its logic from a huge lib.rs file. I think this is nice but can be reverted if you don't like it

Reply:
I do like the modular approach.
I would like @kianenigma or @gupnik to chime in here as to what we would like to see Polkadot developers using. Modular approach or something else?

Reply:
While pallet splitting was a cool experiment and it certainly has its niche use-case, I don't see it is particularly useful for beginner devs as IDE experience is worse, and you might encounter weird compiler issues.

Reply:
Sorry, my comment was a but mislead bc I thought you are using `#[pallet::import_section]` and so on.

Splitting purely into `.rs` files is rather subjective. Similarly, my opinion is that it is better to small with one file rather than few, but I am sure some devs prefer it the other way. 

Contrary, something that worsens the IDE experience (`#[import_section]`) is more objective.

Reply:
I completely understand. If we do move forward with the modular approach, does the current modular setup (the way that it is structured) look good to you?

Reply:
As this is a subjective topic, probably users may tell us if they like one approach or another one after merging. I personally like this one due it moves a lot of logic out of the macro expansion, but also understand the other point :)